cpp/src/sampling/random_walks_impl.cuh

```diff
@@ -221,6 +587,13 @@ random_walk_impl(raft::handle_t const& handle,
                        ? std::make_optional<rmm::device_uvector<weight_t>>(0, handle.get_stream())
                        : std::nullopt;
 
+  auto previous_vertices = (walk_type == random_walk_t::NODE2VEC) 
```

Comment:
It feels like we could use `std::is_same` on `random_selector_t` to determine this and not need to add the `random_walk_t` type at all.  Or if we want more flexibility, you could either:
1. Add a `constexpr` to the `random_selector_t` classes that is true for node2vec and false otherwise that you could use here, or
2. Add a function in `random_selector_t` that returns `std::nullopt`, except in node2vec where it allocates the optional device uvector

Seems like adding an enumerated type just for this one if statement is overkill.